src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.ExtensionMethodSymbolComputer.cs

```diff
@@ -137,6 +144,10 @@ public async Task<(ImmutableArray<IMethodSymbol> symbols, bool isPartialResult)>
                 var browsableSymbols = symbols.ToImmutable()
                     .FilterToVisibleAndBrowsableSymbols(hideAdvancedMembers, _originatingSemanticModel.Compilation);
 
+                // If we are not force creating/updating the cache, an update task needs to be queued in background.
+                if (!forceCacheCreation)
+                    _workQueue.AddWork((_originatingDocument.Project.Solution.Workspace, _originatingDocument.Project.Id));
```

Comment:
why is this the last step, and not the first one?

Reply:
(just trying to understand)

Reply:
No particular reason. I don't see much difference between being first and last step though

Reply:
the difference is effectivley taht the prior lines may exit early (including through cancellation).  I'm trying to figure out if it's an issue if you end up not enqueuing this work here in that case.  Thanks!

Reply:
You are right. Fixed